libs/payments/cart/src/lib/cart.service.ts

```diff
@@ -639,6 +643,31 @@ export class CartService {
             price,
             cartDetails.currency || oldCart.currency
           );
+
+          if (oldCart.stripeCustomerId) {
+            try {
+              const customer = await this.customerManager.retrieve(
+                oldCart.stripeCustomerId
+              );
+              await this.invoiceManager.previewUpcoming({
+                priceId: price.id,
+                currency: cartDetails.currency || oldCart.currency,
+                customer,
+                taxAddress: cartDetails.taxAddress,
+                couponCode: cartDetailsInput.couponCode,
+              });
+            } catch (error) {
+              if (
+                error.type === 'StripeInvalidRequestError' &&
+                error.message ===
+                  'This promotion code cannot be redeemed because the associated customer has prior transactions.'
+              ) {
+                throw new CouponErrorCannotRedeem();
+              } else {
+                throw new CouponErrorInvalid();
```

Comment:
issue: errors caught by this try-catch are not necessarily a coupon related error.

suggestion: throw the original error. i.e. `throw error`